ObjCVoiceCallKitQuickstart/ViewController.m

```diff
@@ -266,12 +268,22 @@ - (void)call:(TVOCall *)call didDisconnectWithError:(NSError *)error {
         NSLog(@"Call disconnected");
     }
 
+    if (!self.userInitiatedDisconnect) {
+        CXCallEndedReason reason = CXCallEndedReasonRemoteEnded;
+        if (error) {
+            reason = CXCallEndedReasonFailed;
+        }
+        
+        [self.callKitProvider reportCallWithUUID:call.uuid endedAtDate:[NSDate date] reason:reason];
+    }
```

Comment:
What happens if you do not check the flag and call [self.callKitProvider reportCallWithUUI:] every time you receive [call:(TVOCall *)call didDisconnectWithError] ?

Is there a way to avoid an extra flag?

Reply:
I tried ending the call locally and report the end-date to CallKit without the flag - didn't see anything printed out indicating warning or error, although the end-call request should have been reported to CallKit in this case.

Reply:
lets not add an extra flag and call `[self.callKitProvider reportCallWithUUID:call.uuid endedAtDate:[NSDate date] reason:reason];` when `didDisconnectWithError` is called.